# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../sun-jre-150/mime-type-remove.diff
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
we don't want java touching /etc/mailcap
Alejandro

we dont want java touching /usr/share/ registry files
mipe

we can't add new lines to .bin file because it does tail to checksum and to
generate tarball
Alejandro

Just a remark without modification: safe_mimetypeadd can not be used to add
mimetypes safely. This is done in postsysfiles.in. Create both mimetype and
mailcap entries.
Minto

--- jdk-1_5_0_03-linux-i586-orig.bin	2005-06-08 17:26:07.000000000 +0000
+++ jdk-1_5_0_03-linux-i586.bin	2005-06-08 17:50:13.000000000 +0000
@@ -356,7 +356,7 @@
     if [ -w `dirname ${MAILCAP_FILE}` ]; then
         MC_TEXT="${MC_TEXT}\n${MC_COMMENT}"
         MC_TEXT="${MC_TEXT}\n${MIME_TYPE}; $LATEST_JAVAWS_PATH %s"
-        echo ${echo_args}"${MC_TEXT}" > ${MAILCAP_FILE}
+        #echo ${echo_args}"${MC_TEXT}" > ${MAILCAP_FILE}
     else
         echo "WARNING - cannot write to file:"
         echo "       ${MAILCAP_FILE}"
@@ -379,11 +379,11 @@
 
     # Create the file if it does not exist
     if [ ! -w ${MIME_FILE} ]; then
-        if [ -w `dirname ${MIME_FILE}` ]; then
-            echo "${NS_COMMENT1}"  > ${MIME_FILE}
-            echo "${NS_COMMENT2}" >> ${MIME_FILE}
-            echo "${NS_COMMENT3}" >> ${MIME_FILE}
-        else
+        if [ ! -w `dirname ${MIME_FILE}` ]; then
+        #    echo "${NS_COMMENT1}"  > ${MIME_FILE}
+        #    echo "${NS_COMMENT2}" >> ${MIME_FILE}
+        #    echo "${NS_COMMENT3}" >> ${MIME_FILE}
+        #else
             echo "WARNING - cannot write to file:"
             echo "       ${MIME_FILE}"
             echo "Check permissions."
@@ -391,9 +391,9 @@
         fi
     fi
     # Add the jnlp entry if it does not already exist.
-    if [ -z "`grep ${MIME_TYPE} ${MIME_FILE}`" ]; then
-        echo ${JNLP_ENTRY} >> ${MIME_FILE}
-    fi
+    #if [ -z "`grep ${MIME_TYPE} ${MIME_FILE}`" ]; then
+    #    echo ${JNLP_ENTRY} >> ${MIME_FILE}
+    #fi
 }
 # End of UpdateMimeTypes
 
@@ -496,32 +496,32 @@
 
         # since we are pretty certain this is a valid GNOME installation, create
         # the required association directories in case they don't already exist.
-        mkdir -p ${_gnome_dir}/${MIME_INFO}
-        mkdir -p ${_gnome_dir}/${APP_REGISTRY}
+        #mkdir -p ${_gnome_dir}/${MIME_INFO}
+        #mkdir -p ${_gnome_dir}/${APP_REGISTRY}
 
         GNOME_KEYS_FILE=${_gnome_dir}/${MIME_INFO}/${_name}.keys
         GNOME_MIME_FILE=${_gnome_dir}/${MIME_INFO}/${_name}.mime
         GNOME_APPS_FILE=${_gnome_dir}/${APP_REGISTRY}/${_name}.applications
 
         # Create the keys file.
-        echo "${_mime_type}"                                           > $GNOME_KEYS_FILE
-        echo "        description=${_description}"                    >> $GNOME_KEYS_FILE
-        echo "        icon_filename=${_icon}"                         >> $GNOME_KEYS_FILE
-        echo "        default_action_type=application"                >> $GNOME_KEYS_FILE
-        echo "        default_application_id=${_name}"                >> $GNOME_KEYS_FILE
-        echo "        short_list_application_user_additions=${_name}" >> $GNOME_KEYS_FILE
+        #echo "${_mime_type}"                                           > $GNOME_KEYS_FILE
+        #echo "        description=${_description}"                    >> $GNOME_KEYS_FILE
+        #echo "        icon_filename=${_icon}"                         >> $GNOME_KEYS_FILE
+        #echo "        default_action_type=application"                >> $GNOME_KEYS_FILE
+        #echo "        default_application_id=${_name}"                >> $GNOME_KEYS_FILE
+        #echo "        short_list_application_user_additions=${_name}" >> $GNOME_KEYS_FILE
 
         # Create the mime file.
-        echo "${_mime_type}"               > $GNOME_MIME_FILE
-        echo "        ext: ${_extension}" >> $GNOME_MIME_FILE
+        #echo "${_mime_type}"               > $GNOME_MIME_FILE
+        #echo "        ext: ${_extension}" >> $GNOME_MIME_FILE
 
         # Create the apps file.
-        echo "${_name}"                               > $GNOME_APPS_FILE
-        echo "        command=${_command}"           >> $GNOME_APPS_FILE
-        echo "        name=${_name}"                 >> $GNOME_APPS_FILE
-        echo "        can_open_multiple_files=false" >> $GNOME_APPS_FILE
-        echo "        requires_terminal=false"       >> $GNOME_APPS_FILE
-        echo "        mime_types=${_mime_type}"      >> $GNOME_APPS_FILE
+        #echo "${_name}"                               > $GNOME_APPS_FILE
+        #echo "        command=${_command}"           >> $GNOME_APPS_FILE
+        #echo "        name=${_name}"                 >> $GNOME_APPS_FILE
+        #echo "        can_open_multiple_files=false" >> $GNOME_APPS_FILE
+        #echo "        requires_terminal=false"       >> $GNOME_APPS_FILE
+        #echo "        mime_types=${_mime_type}"      >> $GNOME_APPS_FILE
     fi
 }
 # End of AddGNOME
@@ -639,21 +639,21 @@
         fi
 
         # Setup the GNOME associations
-        AddGNOME application/java-archive \
-                 jar \
-                 java-archive \
-                 "${LATEST_JRE_PATH}/bin/java -jar" \
-                 sun-java \
-                 png \
-                 "Java Archive"
-
-        AddGNOME ${MIME_TYPE} \
-                 jnlp \
-                 java-web-start \
-                 ${LATEST_JAVAWS_PATH} \
-                 sun-java \
-                 png \
-                 "Java Web Start Application"
+        #AddGNOME application/java-archive \
+        #         jar \
+        #         java-archive \
+        #         "${LATEST_JRE_PATH}/bin/java -jar" \
+        #         sun-java \
+        #         png \
+        #         "Java Archive"
+
+        #AddGNOME ${MIME_TYPE} \
+        #         jnlp \
+        #         java-web-start \
+        #         ${LATEST_JAVAWS_PATH} \
+        #         sun-java \
+        #         png \
+        #         "Java Web Start Application"
     fi
 fi
